### PR TITLE
Add error message if not logged in for adding products to wishlist

### PR DESCRIPTION
**Pull request by PieterCappelle**
### Description

When you add a product to the wishlist and you're not logged in you're redirected to the login page without any warning. In this PR I've added short warning why you're redirect to the login page.

### Manual testing scenarios
Add product to wishlist without logging in.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Wishlist/Controller/Index/Plugin.php [hunk 1]
```diff
@@ -30,26 +30,34 @@ class Plugin
     protected $config;
 
     /**
-     * @var \Magento\Framework\App\Response\RedirectInterface
+     * @var \Magento\Framework\Message\ManagerInterface $messageManager
```
Comment (ishakhsuvarov) [inline app/code/Magento/Wishlist/Controller/Index/Plugin.php]: Looks like this change was done unintentionally.

File: app/code/Magento/Wishlist/Controller/Index/Plugin.php [hunk 2]
```diff
@@ -30,26 +30,34 @@ class Plugin
...
      */
     protected $redirector;
 
+    /**
+     * @var \Magento\Framework\Message\ManagerInterface $messageManager
+     */
+    protected $messageManager;
```
Comment (ishakhsuvarov) [inline app/code/Magento/Wishlist/Controller/Index/Plugin.php]: Please use only `private` properties for core code as usage of inheritance is not encouraged.

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Wishlist/Controller/Index/Plugin.php (hunk 2 above)
Comment (PieterCappelle) [reply app/code/Magento/Wishlist/Controller/Index/Plugin.php]: Done.

File: app/code/Magento/Wishlist/Controller/Index/Plugin.php (hunk 1 above)
Comment (PieterCappelle) [reply app/code/Magento/Wishlist/Controller/Index/Plugin.php]: Done.

**Review by ishakhsuvarov — APPROVED** [review]

**Pull request merged**